tests/unit/test_github_enterprise.py

```diff
@@ -0,0 +1,134 @@
+from github3.github import GitHubEnterprise
+import github3
+
+from .helper import UnitHelper
+
+example_data = {
+    "login": "octocat",
+    "id": 1,
+    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
+    "gravatar_id": "somehexcode",
+    "url": "https://api.github.com/users/octocat",
+    "html_url": "https://github.com/octocat",
+    "followers_url": "https://api.github.com/users/octocat/followers",
+    "following_url": ("https://api.github.com/users/octocat/following"
+                      "{/other_user}"),
+    "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
+    "starred_url": ("https://api.github.com/users/octocat/starred"
+                    "{/owner}{/repo}"),
+    "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
+    "organizations_url": "https://api.github.com/users/octocat/orgs",
+    "repos_url": "https://api.github.com/users/octocat/repos",
+    "events_url": "https://api.github.com/users/octocat/events{/privacy}",
+    "received_events_url": ("https://api.github.com/users/octocat/"
+                            "received_events"),
+    "type": "User",
+    "site_admin": False,
+    "name": "monalisa octocat",
+    "company": "GitHub",
+    "blog": "https://github.com/blog",
+    "location": "San Francisco",
+    "email": "octocat@github.com",
+    "hireable": False,
+    "bio": "There once was...",
+    "public_repos": 2,
+    "public_gists": 1,
+    "followers": 20,
+    "following": 0,
+    "created_at": "2008-01-14T04:33:35Z",
+    "updated_at": "2008-01-14T04:33:35Z"
+}
+
+base_url = 'https://ghe.example.com/'
+
+
+def _build_url(_, *args, **kwargs):
+    return github3.session.GitHubSession().build_url(
+        base_url=base_url + 'api/v3', *args, **kwargs)
```

Comment:
Iterable unpacking and dictionary unpacking can come at the end, and iterable unpacking can come before keyword arguments (and can even be surrounded by them).

```py
>>> def foo(*args, **kwargs):
...     return args, kwargs
...
>>> foo(1, 2, 3, 4, a=1, b=2, c=3)
((1, 2, 3, 4), {'a': 1, 'c': 3, 'b': 2})
>>> args, kwargs = _
>>> foo(*args, **kwargs)
((1, 2, 3, 4), {'a': 1, 'c': 3, 'b': 2})
>>> foo(**kwargs, *args)
  File "<stdin>", line 1
    foo(**kwargs, *args)
                ^
SyntaxError: invalid syntax
>>> foo(*args, d=4, **kwargs)
((1, 2, 3, 4), {'a': 1, 'c': 3, 'b': 2, 'd': 4})
>>> foo(d=4, *args, **kwargs)
((1, 2, 3, 4), {'a': 1, 'c': 3, 'b': 2, 'd': 4})
>>> foo(d=4, *args, e=5, **kwargs)
((1, 2, 3, 4), {'a': 1, 'c': 3, 'b': 2, 'e': 5, 'd': 4})
```

Nothing can come after `**kwargs` though.